src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -155,6 +172,17 @@ public void run() {
      * Returns the file that records the last/current polling activity.
      */
     public File getLogFile() {
+        try {
+            return getLogFileForJob(notNull(job, "Job can't be null!"));
+        } catch (IOException ex) {
+            throw new RuntimeException(ex);
```

Comment:
Why not just add throws to the method and let Jenkins handle it. Like you did with getLogFileForJob.
You're rethrowing the exception just wrapped in another exception. Hence it makes sense to add throws to the method signature.

Reply:
I just didnt want to make a signature change for the method if there were any reverse-compatibility API issues.

Reply:
I was actually tempted to @Deprecate this method, since I don't know how/why it is exposed outside this method

Reply:
This is exposed for Jenkins to provide a view of the trigger log

Reply:
Are you sure? As far as I can tell, it isn't implementing any interface, it compiles/tests fine without it

Reply:
@KostyaSha this public `getLogFile` isn't used anywhere other than the `github-plugin` I would like to simply remove the public interface,

https://github.com/jenkinsci/github-plugin/search?q=getlogfile&unscoped_q=getlogfile
https://github.com/search?q=org%3Ajenkinsci+GitHubPushTrigger+getlogfile&type=Code

As far as I can tell this isn't used by any other project, any objections, or should I simply mark it as `@Deprecated`?

Reply:
jenkinsci is not the only place where plugins exist, there are a lot of private plugins, if it was public it should be public.

Reply:
OK, is the existing code fine, or should I change the signature to throw IOException?